### PR TITLE
feat: create a new snyk view [IDE-711][IDE-712][IDE-713]

### PR DESCRIPTION
**Pull request by acke**
### Description

_Provide description of this PR and changes, if linked Jira ticket doesn't cover it in full._

### Checklist

- [ ] Tests added and all succeed
- [ ] Linted
- [ ] CHANGELOG.md updated
- [ ] README.md updated, if user-facing

### Screenshots / GIFs

_Visuals that may help the reviewer. Please add screenshots for any UI change. GIFs are most welcome!_

### CODE REVIEW

File: plugin/META-INF/MANIFEST.MF [hunk 1]
```diff
@@ -1,7 +1,8 @@
 Manifest-Version: 1.0
 Bundle-ManifestVersion: 2
 Bundle-Name: %Bundle-Name
-Bundle-RequiredExecutionEnvironment: JavaSE-17
+Bundle-RequiredExecutionEnvironment: JavaSE-17,
+ JavaSE-21
```
Comment (bastiandoetsch) [inline plugin/META-INF/MANIFEST.MF:5]: question: why 21? That would be a breaking change :).

File: plugin/plugin.xml [hunk 2]
```diff
@@ -1,271 +1,531 @@
...
+                commandId="io.snyk.eclipse.plugin.commands.snykFilterLow"
+                icon="platform:/plugin/io.snyk.eclipse.plugin/icons/severity-low.png"
+                style="push"
+                tooltip="Low Severity">
+            </command>
+            <command
+                commandId="io.snyk.eclipse.plugin.commands.snykFilterDelta"
+                icon="platform:/plugin/org.eclipse.jdt.ui/icons/full/elcl16/exclusion_filter_attrib.png"
+                style="push"
+                tooltip="Delta findings">
```
Comment (bastiandoetsch) [inline plugin/plugin.xml:257]: shouldn't this be "Net new issues"?

File: plugin/plugin.xml [hunk 3]
```diff
@@ -1,271 +1,531 @@
...
+                relative="org.eclipse.ui.views.ProblemView"
+                relationship="right"
+                ratio="0.5">
+            </view>
+            <view
+                id="io.snyk.eclipse.plugin.views.snyktoolview"
+                relative="org.eclipse.ui.views.ProblemView"
+                relationship="right"
+                ratio="0.5">
+            </view>
```
Comment (bastiandoetsch) [inline plugin/plugin.xml:42]: I'd not automatically add it to the perspective right now, so it does not appear in the Preview automatically without function.

File: plugin/src/main/java/io/snyk/eclipse/plugin/views/SnykView.java [hunk 4]
```diff
@@ -46,359 +46,359 @@
...
+		if (activeWorkbenchWindow == null)
+			return SHELL;
+		return activeWorkbenchWindow.getShell();
+	}
+
+	public void enableScanBasedOnConfig() {
+		try {
+			boolean cliFound = getCliFile().exists();
+			boolean tokenFound = !Preferences.getInstance().getAuthToken().isBlank()
+					|| System.getenv().containsKey("SNYK_TOKEN");
```
Comment (bastiandoetsch) [inline plugin/src/main/java/io/snyk/eclipse/plugin/views/SnykView.java:265]: The SNYK_TOKEN check is old and a bug :)

File: plugin/src/main/java/io/snyk/eclipse/plugin/views/snyktoolview/RootObject.java [hunk 5]
```diff
@@ -0,0 +1,23 @@
+package io.snyk.eclipse.plugin.views.snyktoolview;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.eclipse.jface.viewers.TreeNode;
+
+public class RootObject {
```
Comment (bastiandoetsch) [inline plugin/src/main/java/io/snyk/eclipse/plugin/views/snyktoolview/RootObject.java:8]: I'd prefer `RootNode` as name.

File: plugin/src/main/java/io/snyk/eclipse/plugin/views/snyktoolview/RootObject.java [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+	private List<TreeNode> children;
+
+	public RootObject() {
+		children = new ArrayList<>();
+
+		// Initialize with some data
+		children.add(new TreeNode("Open Source"));
+		children.add(new TreeNode("Code Security"));
```
Comment (bastiandoetsch) [inline plugin/src/main/java/io/snyk/eclipse/plugin/views/snyktoolview/RootObject.java:16]: What about Code Quality?

File: plugin/src/main/java/io/snyk/eclipse/plugin/views/snyktoolview/handlers/EnableCodeProductHandler.java [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+	public Object execute(ExecutionEvent event) throws ExecutionException {
+		String commandId = event.getCommand().getId();
+
+		ICommandService commandService = PlatformUI.getWorkbench().getService(ICommandService.class);
+		if (commandService != null) {
+			commandService.refreshElements(commandId, null);
+		}
+
+		return null;
+	}
```
Comment (bastiandoetsch) [inline plugin/src/main/java/io/snyk/eclipse/plugin/views/snyktoolview/handlers/EnableCodeProductHandler.java:32]: Can't this be pulled up to `AbstractHandler`?

File: plugin/t.xml
Comment (bastiandoetsch) [inline plugin/t.xml:1]: what is this?

File: plugin/src/main/java/io/snyk/eclipse/plugin/views/SnykView.java (hunk 4 above)
Comment (acke) [reply plugin/src/main/java/io/snyk/eclipse/plugin/views/SnykView.java:265]: Should we remove tokenFound, or only the System.getenv().containsKey check?

File: plugin/src/main/java/io/snyk/eclipse/plugin/views/snyktoolview/handlers/EnableCodeProductHandler.java (hunk 7 above)
Comment (acke) [reply plugin/src/main/java/io/snyk/eclipse/plugin/views/snyktoolview/handlers/EnableCodeProductHandler.java]: 100%

**Review by bastiandoetsch — CHANGES_REQUESTED** [review]
Leaving this in "Request Changes" as we'll use this as a base branch/PR for stacked PRs. Once we're ready, I'll move it to Approve :).

**Review by bastiandoetsch — APPROVED** [review]

**Pull request merged**